app/controllers/api/v1/migrations/external_controller.rb

```diff
@@ -38,17 +38,17 @@ def create_role
         # Does: Creates a user.
 
         def create_user
-          language = if user_params[:language].blank? || user_params[:language] == 'default'
-                       I18n.default_locale
-                     else
-                       user_params[:language]
-                     end
+          user_hash = user_params.to_h
 
-          role = Role.find_by(name: user_params[:role], provider: 'greenlight')
+          user_hash[:language] = I18n.default_locale if user_hash[:language].blank? || user_hash[:language] == 'default'
+
+          role = Role.find_by(name: user_hash[:role], provider: 'greenlight')
 
           return render_error status: :bad_request unless role
 
-          user = User.new(user_params.merge(provider: 'greenlight', password: generate_secure_pwd, language:, role:))
+          user_hash[:password] = generate_secure_pwd if user_hash[:external_id].blank?
```

Comment:
We'd like our migrated users from to be conform with their regular created counterparts.
So, we create local migrated users a random password but not our external migrated users,